tensorboard/plugins/debugger_v2/debugger_v2_plugin.py

```diff
@@ -277,3 +279,20 @@ def serve_stack_frames(self, request):
             )
         except errors.NotFoundError as e:
             return _error_response(request, str(e))
+
+    @wrappers.Request.application
+    def serve_dtypes_map(self, request):
```

Comment:
did you think about the alternative? It is a bit weird for backend to regularly flush down the map like this. I can see two other designs:

1. transform other endpoints that flushes the dtype enum in the payload to the right value
2. frontend has this map; if we had jspb and used protobuf everywhere, this would have been very natural place to do the conversion, anyways.

Reply:
Thanks for the suggestion! I think your option 1 makes sense. I can make changes in this PR, so that the endpoint serves debug tensor values are served as a JSON object such as 

```
{
  "tensor_id": 123,
  "dtype": "float32",
  "rank": 2,
  "size": 6
}
```
instead of a simple array such as `[123, 1, 2, 6]` from the end point.

This is going to be a pretty large change. But as you suggested, it's going to make the front-end code simpler and will get rid of the "/dtypes_map" endpoint. 

Let me know what you think.

Reply:
If you think that will bloat the response beyond reasonable bounds, do consider the second option, too. protos do not change anyways.

Reply:
Option 2 will necessitate hard-coding of the dtypes in typescript (either by creating a new .ts file or using genrules).

I think option 1 you suggested is the most elegant
1. It doesn't at to the complexity of the build system
2. It simplifies the FE
3. The increase in response size is acceptible, because the size of the execution/data is protected against thanks to execution/digest anyway.

Reply:
@stephanwlee After some further thinking, I decided that the current approach is reasonable and more appropriate compared to the two options you mentioned. The reason is as follows:

1. The human-readable string names of the dtypes (e.g., "float16") are not a part of the protobuf type definition it self (which is at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/types.proto#L12). The string names of the proto types (e.g., "DT_FLOAT") is not the most human-friendly to show. Instead, the human-readable strings are defined at the Python level, at https://github.com/tensorflow/tensorboard/blob/master/tensorboard/compat/tensorflow_stub/dtypes.py#L480.

As such, it is natural to populate the map from Python. Doing so also leaves open the possibility of attaching additional properties of the types in the response in the future, e.g., including fields such as "is_floating", "is_integer", and "is_string", which are also not pieces of information directly available from the protobuf definition.

This is argument against option 2 above.

2. With regard to option 1 (encoding the semantics, e.g., "dtype" and "num_nans") in the response itself. This will cause the response format to deviate from the on-disk format for the execution data and therefore increase the complexity of the backend. Increase in the size of the response is another minor concern, as mentioned above. 

Based on these considerations, I'd like to stick with the current approach. In the meantime, I've slightly optimized the Python code in `debugger_v2_plugin.py`, so that the `dtypes_map` is populated once, when the plugin is created, instead of being created on the fly during every request.

PTAL.